### ISSUE TITLE
Adding a new store

### ISSUE THREAD

**Issue by bgriffy** [amuste/DnetIndexedDb#47]
I am not able to add a new store to an existing database. The only way I'm able to add the store is by deleting the whole database. 

From what I understand, in order to do this, I would need to manually increment the version # on the DB which would then force the "UpgradeNeeded" event to fire off.  I do this, the UpgradeDB function appears to execute correctly, but then the next time I go to open the DB I get the following error: 

"Unhandled exception rendering component: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing."

Does anyone else get this error or is there something I'm missing here?

**Comment by shahbaz0001** [comment]
Any update on this ?

I read release note of latest version (2.3.1) and in that there is message  "Bug: Could not add new object store to existing database."
However I am not able to do it ?


![image](https://user-images.githubusercontent.com/48522590/130730973-b9672a76-9768-486e-b98c-1895f9319e29.png)


can anyone please help me on this ?
Thanks

**Comment by bgriffy** [comment]
I am going to close this issue because I've figured out what was causing it for me. 

Whenever I was testing this issue, I would increment the version # in the "IndexedDbInterop.js" file, 

![image](https://user-images.githubusercontent.com/11038998/130966231-acb8a521-9131-4848-bd00-0155c8ec553e.png)

But, this will not work because of the way the _indexedDbDatabaseModel object is scoped. The next time this function is called, the version # will be reverted back to the previous version. 

Instead, what I needed to do was manually increment the version #  when initializing the database model: 

![image](https://user-images.githubusercontent.com/11038998/130966797-63615575-fcd7-4ec3-bb47-31e29856b1c3.png)

By incrementing this value, it will allow you to add a new store to an existing database. Obviously having to manually update the version # each time a new store is added isn't practical, so the versioning should be automated in your apps build process. I.E. retrieve the version # from an app manifest or something that gets set by your build process.
